spacy/language.py

```diff
@@ -1508,6 +1512,8 @@ def from_config(
                     nlp.add_pipe(source_name, source=source_nlps[model], name=pipe_name)
         nlp.config = filled if auto_fill else config
         nlp.resolved = resolved
+        if after_init is not None:
+            after_init(nlp)
```

Comment:
Will things break if the callback wants to return a different object? I don't think we rely on object identity for the `nlp` object anywhere outside here, so I think this will work?

```suggestion
            nlp = after_init(nlp)
```

If it works I would prefer this, as it may allow some situations that would otherwise be difficult to implement.

Reply:
Ah, good point. I think I might have to adjust the type signature for the callback as well then.